pkg/api-server/resource_endpoints.go

```diff
@@ -424,7 +424,7 @@ func (r *resourceEndpoints) createResource(
 			log.Error(err, "Could not write the response")
```

Comment:
IMO This is odd, why don't we just return an empty json instead of the different status code? Both are breaking changes to the API but one seems least surprising (didn't even know 204 existed)

Reply:
You can see that was my first approach, then I wanted to check how 204 would look like and how many places it would break to have some sort of a proxy "metric" for the clients. I agree, empty json will be better.